### ISSUE TITLE
Keras evaluate method returns wrong value after loading a model

### ISSUE THREAD

**Issue by HKatoo** [tensorflow/tensorflow#41958]
**System information**
- OS Platform : Windows10
- TensorFlow installed from : source(pip in anaconda)
- TensorFlow version : 2.3.0
- Python version : 3.8.3

**Describe the current behavior**
Keras Sequential method:`evaluate()` returns wrong value after loading a model. value of loss is equal but metrics differs before and after the model is saved and loaded.
I calculated the accuracy manually using sklearn.metrics and got the same value. Therefore, the problem seems to be in this method.
In the below code, the `acc1`, `acc2` are the same. However `acc1=0.863` and `acc2=0.086` in my environment.

~~When I ran the code in colab, this issue was not occurred.~~

**Describe the expected behavior**
Keras Sequential method:`evaluate()` returns same value before and after the model is saved and loaded.

**Standalone code to reproduce the issue**
This code is a modified version of this [tutorial](https://www.tensorflow.org/tutorials/keras/save_and_load) code to reproduce the issue.
If you run the code in the above tutorial on Windows10, I think you are able to reproduce the issue.
For your information, I comment on the results in my environment.

```
import numpy as np
import tensorflow as tf
from tensorflow import keras
from sklearn.metrics import accuracy_score

(train_images, train_labels),(test_images, test_labels) = tf.keras.datasets.mnist.load_data()
train_labels = train_labels[:1000]
test_labels = test_labels[:1000]
train_images = train_images[:1000].reshape(-1, 28 * 28) / 255.0
test_images = test_images[:1000].reshape(-1, 28 * 28) / 255.0

def create_model():
    model = tf.keras.models.Sequential([
        keras.layers.Dense(512, activation='relu', input_shape=(784,)),
        keras.layers.Dropout(0.2),
        keras.layers.Dense(10)
    ])
    model.compile(optimizer='adam',
                  loss=tf.losses.SparseCategoricalCrossentropy(from_logits=True),
                  metrics=['accuracy'])
    return model

# Create first model
model = create_model()
model.fit(train_images, train_labels, epochs=5)

model.save('saved_model/my_model')

# Create second model (load the saved model)
new_model = tf.keras.models.load_model('saved_model/my_model')

# First model's results
loss1, acc1 = model.evaluate(test_images, test_labels)
accuracy_score1 = accuracy_score(test_labels, np.argmax(model.predict(test_images), axis=1))
# loss1, acc1 = [0.4392167329788208, 0.8629999756813049]
# accuracy_score1 = 0.863

# Second model's results
loss2, acc2 = new_model.evaluate(test_images, test_labels)                           
accuracy_score2 = accuracy_score(test_labels, np.argmax(new_model.predict(test_images), axis=1))
# loss2, acc2 = [0.4392167329788208, 0.0860000029206276] <- THIS!!!
# accuracy_score2 = 0.863
```

Why is this happening?
Can you help me?

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with TF v2.2 and TF v2.3. Please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/bcbb82c4dc189486a4bd7594bfbe92f2/41958.ipynb). Thanks!

**Comment by jvishnuvardhan** [comment]
@HKatoo This is a known issue that team is working on. Please check the the comment by @k-w-w 

> This is a bug with using the sparse categorical accuracy. For now, please compile the model with metrics='sparse_categorical_accuracy' instead of just 'accuracy'.

With the above modification, `acc1` and `acc2` are the same. Please check the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/c50083ed3bdeea12ef14ef83724e6cbb/41958.ipynb).

Let's monitor the progress [here](https://github.com/tensorflow/tensorflow/issues/42045#issuecomment-674232499) in this issue. Thanks!

**Comment by HKatoo** [comment]
@jvishnuvardhan Thank you!

**Comment by jvishnuvardhan** [comment]
This is resolved in recent tf-nightly. Please feel free to reopen if you notice the issue.
This is available in stable TF2.4 in the near future. Thanks!
